### PR TITLE
Added debounce feature to Filters and also added fix for clearing filter

### PR DESCRIPTION
**Pull request by maheshkulkarni4**
https://jira.cec.lab.emc.com/browse/OBSDEF-4746

1) Added Debounce of 500ms

Filters on ObjectStoreList, events, issues pages fires too many backend calls, as every keystroke fires one backend call. This is not really performant, so to reduce this behaviour we have added debounce with 500ms to all the text based filters.

2) DataGridFilter didnt had option to clear themself, so exposed a method to clear the dataGrid Filter by passing Empty string

3) Introducted **disabled** prop to allow disabling filter textbox if the props is applied. This is important to avoid user typing into the textbox when some backend calls are already in progress and loader is visible.  In current implementation, the DataGrid Loader does not block user from keep typing into filter.  So this disabled prop would help solve this issue.

### CODE REVIEW

File: src/datagrid/DataGridFilter.tsx [hunk 1]
```diff
@@ -251,9 +264,9 @@ export class DataGridFilter extends React.PureComponent<DataGridFilterProps, Dat
     }
 
     render() {
-        const {isOpen} = this.state;
+        const {isOpen, isFiltered} = this.state;
         const {showFilter} = this.props;
-        const {isFiltered} = this;
+        //const {isFiltered} = this;
```
Comment (girishdudhe) [inline src/datagrid/DataGridFilter.tsx:269]: remove commented line

**Review by girishdudhe — APPROVED** [review]
LGTM ...

File: src/datagrid/DataGridFilter.tsx [hunk 2]
```diff
@@ -237,9 +249,10 @@ export class DataGridFilter extends React.PureComponent<DataGridFilterProps, Dat
                             name={`name-${columnName}`}
                             placeholder={placeholder}
                             defaultValue={filterValue}
-                            onChange={evt => {
-                                this.updateFilter(evt.target.value);
-                            }}
+                            disabled={disabled || false}
+                            onChange={(evt: React.ChangeEvent<HTMLInputElement>) =>
```
Comment (akate1) [inline src/datagrid/DataGridFilter.tsx:253]: can we add a prop to enable/disable `debounce `effect ?

**Review by akate1 — APPROVED** [review]
LGTM

File: src/datagrid/DataGridFilter.tsx [hunk 3]
```diff
@@ -129,6 +136,11 @@ export class DataGridFilter extends React.PureComponent<DataGridFilterProps, Dat
         }
     }
 
+    private handlOnChange = (evt: React.ChangeEvent<HTMLInputElement>) => {
```
Comment (psivakumaran) [inline src/datagrid/DataGridFilter.tsx:139]: Minor comment 
handlChange -> handleChange

**Review by psivakumaran — APPROVED** [review]
LGTM.

File: src/datagrid/DataGridFilter.tsx (hunk 2 above)
Comment (maheshkulkarni4) [reply src/datagrid/DataGridFilter.tsx:266]: Added now. I dont expect to users to use it.  Default 500ms is good for standard usecases. But good to have to have option if we want to disable it.

File: src/datagrid/DataGridFilter.tsx (hunk 3 above)
Comment (maheshkulkarni4) [reply src/datagrid/DataGridFilter.tsx]: Addressed.

File: src/datagrid/DataGridFilter.tsx (hunk 1 above)
Comment (maheshkulkarni4) [reply src/datagrid/DataGridFilter.tsx]: Addressed

**Pull request merged**